registration-processor/core-processor/registration-processor-bio-dedupe-stage/src/test/java/io/mosip/registration/processor/biodedupe/stage/BioDedupeProcessorTest.java

```diff
@@ -85,604 +91,833 @@
+        ArgumentCaptor<InternalRegistrationStatusDto> argumentCaptor = ArgumentCaptor.forClass(InternalRegistrationStatusDto.class);
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+        Mockito.verify(registrationStatusService).updateRegistrationStatus(argumentCaptor.capture(),anyString(),anyString());
+        assertTrue(messageDto.getIsValid());
+        assertFalse(messageDto.getInternalError());
+        Assert.assertEquals(argumentCaptor.getValue().getStatusCode(), RegistrationStatusCode.FAILED.name());
+    }
+
+    /*
+        use case 6 -> More than one match of RID/AID found at bio dedupe stage, all duplicate matches possess different UINs.
+        Among that one UIN matches with update packet UIN
+        This is not a valid use case. because abisHandlerUtil.getUniqueRegIds() will not return rid's associated with current packet.
+    */
+
+    /*
+        use case 7 -> More than one match of RID/AID found at bio dedupe stage, all duplicate matches possess different UINs.
+        Among that no UIN matches with update packet UIN.
+        abisHandlerUtil.getUniqueRegIds() returns the latest rids associated with uin's other than current packet uin.
+   */
+    @Test
+    public void moreThenOneMatchFoundWithDifferentUin() throws ApisResourceAccessException, IOException,
+            PacketDecryptionFailureException, io.mosip.kernel.core.exception.IOException, JsonProcessingException, PacketManagerException {
+        registrationStatusDto.setRegistrationId("27847657360002520190320095014");
+        registrationStatusDto.setRegistrationType("UPDATE");
+        registrationStatusDto.setStatusCode("PROCESSING");
+        Mockito.when(registrationStatusService.getRegistrationStatus(any(),any(),any(), any())).thenReturn(registrationStatusDto);
+        Mockito.when(abisHandlerUtil.getPacketStatus(any()))
+                .thenReturn(AbisConstant.POST_ABIS_IDENTIFICATION);
+
+        Set<String> matchedRidList = new HashSet<>();
+        matchedRidList.add("27847657360002520190320095013");
+        matchedRidList.add("27847657360002520190320095012");
+        Mockito.when(abisHandlerUtil.getUniqueRegIds(any(), any(), anyInt(), any(), any())).thenReturn(matchedRidList);
+        Mockito.when(idRepoService.getUinByRid(anyString(),any()))
+                .thenReturn("9403107397")
+                .thenReturn("9403107399");
+        ArgumentCaptor<InternalRegistrationStatusDto> argumentCaptor = ArgumentCaptor.forClass(InternalRegistrationStatusDto.class);
+        MessageDTO messageDto = bioDedupeProcessor.process(dto, stageName);
+        Mockito.verify(registrationStatusService).updateRegistrationStatus(argumentCaptor.capture(),anyString(),anyString());
+        assertTrue(messageDto.getIsValid());
+        assertFalse(messageDto.getInternalError());
+        Assert.assertEquals(argumentCaptor.getValue().getStatusCode(), RegistrationStatusCode.FAILED.name());
```

Comment:
No. If rid is in PROCESSING state it will be marked to go to Manual Adjudication stage before checking  biometricsUpdateSingleMatchAutoReject.

Reply:
No. If rid is in PROCESSING state then it will marked as send to manual adjudication stage before checking for biometricsUpdateSingleMatchAutoReject.

Reply:
Unit test cases are written not exactly on the code flow, it is to ensure the behavior of the class and it should fail when someone is changing the code later point in time.

In this case, only when biometricsUpdateSingleMatchAutoReject is set to true, there is variation in the behavior of whether to autoreject or send to manual adjudication.

Reply:
Resolving it, for now because this review comment is just on the unit test case, this needs to be discussed later, @Khuddusshariff0022 can we create a task?